Issues/Bugs on initial look

First of all, thank you for opening up this tool, my team is excited to utilize it. We've been utilizing dbt for a few months now, and additional testing is great :)

Two issues that came up while setting up.
First, `dataset` & `schema` seem to be used interchangeably for bigquery profiles. We had dataset, would prefer to keep as dataset, but switched to resolve
```
root@ab14a47c8eeb:/src/app# dbt-dry-run cdm-editor --profiles-dir . --manifest-path target/manifest.json
Traceback (most recent call last):
  File "/usr/local/bin/dbt-dry-run", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/dbt_dry_run/__main__.py", line 5, in main
    exit(cli.run())
  File "/usr/local/lib/python3.8/site-packages/dbt_dry_run/cli.py", line 73, in run
    profiles = read_profiles(parsed_args.profiles_dir)
  File "/usr/local/lib/python3.8/site-packages/dbt_dry_run/cli.py", line 66, in read_profiles
    all_profiles[name] = Profile(**profile)
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for Profile
outputs -> demo -> schema
  field required (type=value_error.missing)
  ```
  Second issue, dbt-dry-run didn't work with [as_number](https://docs.getdbt.com/reference/dbt-jinja-functions/as_number)  in profiles. Work around of hard coding for now.
  ```root@ab14a47c8eeb:/src/app# dbt-dry-run cdm-editor --profiles-dir . --manifest-path target/manifest.json
Traceback (most recent call last):
  File "/usr/local/bin/dbt-dry-run", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/dbt_dry_run/__main__.py", line 5, in main
    exit(cli.run())
  File "/usr/local/lib/python3.8/site-packages/dbt_dry_run/cli.py", line 73, in run
    profiles = read_profiles(parsed_args.profiles_dir)
  File "/usr/local/lib/python3.8/site-packages/dbt_dry_run/cli.py", line 61, in read_profiles
    template = template_env.get_template(PROFILE_FILENAME)
  File "/usr/local/lib/python3.8/site-packages/jinja2/environment.py", line 883, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/usr/local/lib/python3.8/site-packages/jinja2/environment.py", line 857, in _load_template
    template = self.loader.load(self, name, globals)
  File "/usr/local/lib/python3.8/site-packages/jinja2/loaders.py", line 127, in load
    code = environment.compile(source, name, filename)
  File "/usr/local/lib/python3.8/site-packages/jinja2/environment.py", line 638, in compile
    self.handle_exception(source=source_hint)
  File "/usr/local/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/usr/local/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "./profiles.yml", line 14, in template
    maximum_bytes_billed: "{{ env_var('DBT_MAXIMUM_BYTES_BILLED', '250000000000') | as_number }}"
jinja2.exceptions.TemplateAssertionError: no filter named 'as_number'
```
Are these to be expected? I can take a stab at opening a PR to fix at least the first issue :)

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,13 @@
 ## Changelog
 
+## dbt-dry-run v0.3.0 (NOT RELEASED)
+
+### Improvements & Bugfixes
+
+- Added support for `as_number` filter in `profiles.yml`
+
+- Support both `schema` and `dataset` in `profiles.yml` as in BigQuery they are used interchangeably
+
 ## dbt-dry-run v0.2.0
 
 ### Improvements & Bugfixes

--- a/dbt_dry_run/cli.py
+++ b/dbt_dry_run/cli.py
@@ -7,6 +7,7 @@ import yaml
 
 from dbt_dry_run.execution import dry_run_manifest
 from dbt_dry_run.models import Manifest, Profile
+from dbt_dry_run.models.profile import read_profiles
 from dbt_dry_run.result_reporter import ResultReporter
 
 parser = argparse.ArgumentParser(description="Dry run DBT")
@@ -41,35 +42,21 @@ parser.add_argument("--report-path", type=str, help="Json path to dump report to
 PROFILE_FILENAME = "profiles.yml"
 
 
-def profiles_get_env_var(key: str, default: str = None) -> Optional[str]:
-    return os.environ.get(key, default)
-
-
-def read_profiles(path: str) -> Dict[str, Profile]:
-    all_profiles: Dict[str, Profile] = {}
-
+def read_profiles_file(path: str) -> Dict[str, Profile]:
     profile_filepath = os.path.join(path, PROFILE_FILENAME)
     if not os.path.exists(profile_filepath):
         raise FileNotFoundError(
             f"Could not find '{PROFILE_FILENAME}' at '{profile_filepath}'"
         )
-
-    template_loader = jinja2.FileSystemLoader(searchpath=path)
-    template_env = jinja2.Environment(loader=template_loader)
-    template_env.globals.update(env_var=profiles_get_env_var)
-    template = template_env.get_template(PROFILE_FILENAME)
-    output_text = template.render()
-    profile_data = yaml.safe_load(output_text)
-    for name, profile in profile_data.items():
-        if name != "config":
-            all_profiles[name] = Profile(**profile)
-    return all_profiles
+    with open(profile_filepath) as f:
+        file_contents = f.read()
+    return read_profiles(file_contents)
 
 
 def run() -> int:
     parsed_args = parser.parse_args()
     manifest = Manifest.from_filepath(parsed_args.manifest_path)
-    profiles = read_profiles(parsed_args.profiles_dir)
+    profiles = read_profiles_file(parsed_args.profiles_dir)
     try:
         profile = profiles[parsed_args.profile]
     except KeyError:

--- a/dbt_dry_run/models/profile.py
+++ b/dbt_dry_run/models/profile.py
@@ -1,7 +1,11 @@
+import os
+from ast import literal_eval
 from enum import Enum
 from pathlib import Path
 from typing import Any, Dict, List, Optional
 
+import jinja2
+import yaml
 from pydantic import Field, root_validator
 from pydantic.main import BaseModel
 
@@ -15,13 +19,22 @@ class Output(BaseModel):
     output_type: str = Field(..., alias="type")
     method: BigQueryConnectionMethod
     project: str
-    db_schema: str = Field(..., alias="schema")
+    db_schema: Optional[str] = Field(None, alias="schema")
+    dataset: Optional[str] = None
     location: str
     threads: int = Field(..., ge=1)
     timeout_seconds: int = Field(..., ge=0)
     keyfile: Optional[Path] = None
     impersonate_service_account: Optional[str] = None
     scopes: List[str] = []
+
+    @property
+    def dataset_or_schema(self) -> str:
+        dataset = self.db_schema or self.dataset
+        if dataset:
+            return dataset
+        else:
+            raise ValueError(f"Must specify `schema` or `dataset` in BigQuery profile")
 
 
 class Profile(BaseModel):
@@ -37,3 +50,27 @@ class Profile(BaseModel):
                 f"target={target} but it must be valid output={output_keys}"
             )
         return values
+
+
+def profiles_get_env_var(key: str, default: str = None) -> Optional[str]:
+    return os.environ.get(key, default)
+
+
+def as_number_filter(value: str) -> Any:
+    return literal_eval(value)
+
+
+def read_profiles(profile_string: str) -> Dict[str, Profile]:
+    all_profiles: Dict[str, Profile] = {}
+
+    template_loader = jinja2.DictLoader({"profiles.yml": profile_string})
+    template_env = jinja2.Environment(loader=template_loader)
+    template_env.globals.update(env_var=profiles_get_env_var)
+    template_env.filters["as_number"] = as_number_filter
+    template = template_env.get_template("profiles.yml")
+    output_text = template.render()
+    profile_data = yaml.safe_load(output_text)
+    for name, profile in profile_data.items():
+        if name != "config":
+            all_profiles[name] = Profile(**profile)
+    return all_profiles

--- a/dbt_dry_run/models/profile.py
+++ b/dbt_dry_run/models/profile.py
@@ -19,7 +19,6 @@ class Output(BaseModel):
     output_type: str = Field(..., alias="type")
     method: BigQueryConnectionMethod
     project: str
-    db_schema: Optional[str] = Field(None, alias="schema")
     dataset: Optional[str] = None
     location: str
     threads: int = Field(..., ge=1)
@@ -28,13 +27,14 @@ class Output(BaseModel):
     impersonate_service_account: Optional[str] = None
     scopes: List[str] = []
 
-    @property
-    def dataset_or_schema(self) -> str:
-        dataset = self.db_schema or self.dataset
-        if dataset:
-            return dataset
-        else:
-            raise ValueError(f"Must specify `schema` or `dataset` in BigQuery profile")
+    @root_validator(pre=True)
+    def validate_dataset_or_schema(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        if values.get("schema") and values.get("dataset"):
+            raise ValueError(f"Must specify one of dataset or schema")
+        elif not values.get("schema") and not values.get("dataset"):
+            raise ValueError(f"Must specify dataset or schema")
+        values["dataset"] = values.get("dataset", values.get("schema"))
+        return values
 
 
 class Profile(BaseModel):

--- a/dbt_dry_run/test/models/test_profile.py
+++ b/dbt_dry_run/test/models/test_profile.py
@@ -1,5 +1,7 @@
 import os
 
+import pytest
+
 from dbt_dry_run.models.profile import read_profiles
 
 
@@ -62,9 +64,7 @@ def test_dataset_schema_alias() -> None:
     """
 
     profiles = read_profiles(profile_string)
-    assert (
-        profiles["default"].outputs["test-output"].dataset_or_schema == "dry_run_schema"
-    )
+    assert profiles["default"].outputs["test-output"].dataset == "dry_run_schema"
 
     profile_string = """
     default:
@@ -82,7 +82,23 @@ def test_dataset_schema_alias() -> None:
     """
 
     profiles = read_profiles(profile_string)
-    assert (
-        profiles["default"].outputs["test-output"].dataset_or_schema
-        == "dry_run_dataset"
-    )
+    assert profiles["default"].outputs["test-output"].dataset == "dry_run_dataset"
+
+    profile_string = """
+    default:
+        target: test-output
+
+        outputs:
+            test-output:
+              type: bigquery
+              method: oauth
+              project:  my_project
+              dataset: dry_run_dataset
+              schema: dry_run_schema
+              location: "{{ env_var('DBT_DRY_RUN_TEST_LOCATION') }}"
+              threads: 4
+              timeout_seconds: 300
+    """
+
+    with pytest.raises(ValueError):
+        read_profiles(profile_string)

--- a/dbt_dry_run/test/models/test_profile.py
+++ b/dbt_dry_run/test/models/test_profile.py
@@ -1,0 +1,88 @@
+import os
+
+from dbt_dry_run.models.profile import read_profiles
+
+
+def test_as_number_filter() -> None:
+    profile_string = """
+    default:
+        target: test-output
+
+        outputs:
+            test-output:
+              type: bigquery
+              method: oauth
+              project:  my_project
+              schema: dry_run
+              location: EU
+              threads: "{{ '8' | as_number }}"
+              timeout_seconds: 300
+    """
+
+    profiles = read_profiles(profile_string)
+    assert profiles["default"].outputs["test-output"].threads == 8
+
+
+def test_env_var_filter() -> None:
+    expected_location = "LOCATION_TEST"
+    os.environ["DBT_DRY_RUN_TEST_LOCATION"] = expected_location
+    profile_string = """
+    default:
+        target: test-output
+
+        outputs:
+            test-output:
+              type: bigquery
+              method: oauth
+              project:  my_project
+              schema: dry_run
+              location: "{{ env_var('DBT_DRY_RUN_TEST_LOCATION') }}"
+              threads: 4
+              timeout_seconds: 300
+    """
+
+    profiles = read_profiles(profile_string)
+    assert profiles["default"].outputs["test-output"].location == expected_location
+
+
+def test_dataset_schema_alias() -> None:
+    profile_string = """
+    default:
+        target: test-output
+
+        outputs:
+            test-output:
+              type: bigquery
+              method: oauth
+              project:  my_project
+              schema: dry_run_schema
+              location: "{{ env_var('DBT_DRY_RUN_TEST_LOCATION') }}"
+              threads: 4
+              timeout_seconds: 300
+    """
+
+    profiles = read_profiles(profile_string)
+    assert (
+        profiles["default"].outputs["test-output"].dataset_or_schema == "dry_run_schema"
+    )
+
+    profile_string = """
+    default:
+        target: test-output
+
+        outputs:
+            test-output:
+              type: bigquery
+              method: oauth
+              project:  my_project
+              dataset: dry_run_dataset
+              location: "{{ env_var('DBT_DRY_RUN_TEST_LOCATION') }}"
+              threads: 4
+              timeout_seconds: 300
+    """
+
+    profiles = read_profiles(profile_string)
+    assert (
+        profiles["default"].outputs["test-output"].dataset_or_schema
+        == "dry_run_dataset"
+    )

